systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaTopicScalabilityUtils.java

```diff
@@ -0,0 +1,127 @@
+        LOGGER.info("Creating topics via Kubernetes");
+
+        for (int i = 0; i < numberOfTopics; i++) {
+            String currentTopicName = topicPrefix + i;
+            LOGGER.debug("Creating topic {}", currentTopicName);
+
+            CompletableFuture.runAsync(() ->
+                    resourceManager.createResource(extensionContext, false, KafkaTopicTemplates.topic(
+                            clusterName, currentTopicName, numberOfPartitions, numberOfReplicas, minInSyncReplicas, INFRA_NAMESPACE).build()));
+        }
+    }
+
+    public static void checkTopicsState(String topicPrefix, int numberOfTopics, Enum<?> state) {
+        List<CompletableFuture<?>> topics = new ArrayList<>();
+
+        for (int i = 0; i < numberOfTopics; i++) {
+            String currentTopic = topicPrefix + i;
+            topics.add(CompletableFuture.runAsync(() ->
+                    KafkaTopicUtils.waitForKafkaTopicStatus(INFRA_NAMESPACE, currentTopic, state)
+            ));
+        }
+
+        CompletableFuture<Void> allTopics = CompletableFuture.allOf(topics.toArray(new CompletableFuture[0]))
+                .thenRun(() -> LOGGER.info("All topics are in correct state"));
+
+        allTopics.join();
+    }
+
+    public static void checkTopicsNotReady(String topicPrefix, int numberOfTopics) {
+        LOGGER.info("Verifying that topics are in NotReady state");
+        KafkaTopicScalabilityUtils.checkTopicsState(topicPrefix, numberOfTopics, CustomResourceStatus.NotReady);
+    }
+
+    public static void checkTopicsReady(String topicPrefix, int numberOfTopics) {
+        LOGGER.info("Verifying that topics are in Ready state");
+        KafkaTopicScalabilityUtils.checkTopicsState(topicPrefix, numberOfTopics, CustomResourceStatus.Ready);
+    }
+
+    public static void checkTopicsContainConfig(String topicPrefix, int numberOfTopics, Map<String, Object> config) {
+        LOGGER.info("Verifying that topics contain right config");
+        List<CompletableFuture<?>> topics = new ArrayList<>();
+
+        for (int i = 0; i < numberOfTopics; i++) {
+            String currentTopic = topicPrefix + i;
+            topics.add(CompletableFuture.runAsync(() -> {
+                try {
+                    String json = cmdKubeClient(kubeClient().getNamespace()).exec("get", "kafkatopic", currentTopic, "-o", "jsonpath='{.spec.config}'").out();
+                    HashMap<?, ?> mapping = new ObjectMapper().readValue(json.replaceAll("'", ""), HashMap.class);
+                    assertEquals(mapping, config);
+                } catch (JsonProcessingException e) {
+                    fail(e.getMessage());
+                }
+            }));
+        }
+
+        CompletableFuture<Void> allTopics = CompletableFuture.allOf(topics.toArray(new CompletableFuture[0]))
+                .thenRun(() -> LOGGER.info("All topics contain right config"));
+
+        allTopics.join();
+    }
```

Comment:
Here you can hit race condition, where one (or more) topics will not have correct configuration at the time you are checking it. It would be better to have here wait for the config to change.